sparse/src/KokkosSparse_Utils_mkl.hpp

```diff
@@ -63,7 +63,7 @@ inline void mkl_internal_safe_call(sparse_status_t mkl_status, const char *name,
 }
 
 #define KOKKOSKERNELS_MKL_SAFE_CALL(call) \
-  KokkosSparse::Impl::mkl_internal_safe_call(call, #call, __FILE__, __LINE__)
+  ::KokkosSparse::Impl::mkl_internal_safe_call(call, #call, __FILE__, __LINE__)
```

Comment:
```suggestion
  mkl_internal_safe_call(call, #call, __FILE__, __LINE__)
```

Reply:
I'm not sure. I agree that the macro is defined here in a piece of code inside the `KokkosSparse::Impl` namespace. But, then, when the macro is used somewhere else, this namespace is not taken with it. 

E.g. when it's used here

- https://github.com/kokkos/kokkos-kernels/blob/afd65f03a52e4d9d2536feb9be93150bdb8dec37/sparse/src/KokkosSparse_spgemm_handle.hpp#L240-L242

it's called in a piece of code in the `KokkosSparse` namespace without the `Impl`. So shouldn't we keep the proposed change?

Reply:
Good point this would obviously create issues. Would you mind moving the definition of the macro `KOKKOSKERNELS_MKL_SAFE_CALL` to line 232, just after the namespaces are closed?
My suspicion is at that point the leading `::` in your fix might not be needed anymore?